### ISSUE TITLE
[Bug][a11y]: Toast should have API for overriding the icon's alt text

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3406]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Toast

### Expected behavior

It should be possible to override the string used as the `label` for the `sp-icon` element in the Shadow DOM, both for i18n purposes and for clarity, because sometimes an "Warning" is not an "Error."

This relates to https://github.com/adobe/spectrum-web-components/issues/1975

### Actual behavior

The alt text for the icon displayed in the "negative", "positive", and "info" Toast variants is hardcoded and cannot be overridden as "Error", "Success", and "Information" respectively.

### How can we reproduce this issue?

3. Scroll to '....'
4. Check console
5. See error
